src/cursor/cur_index.c

```diff
@@ -203,6 +203,13 @@ __curindex_reset(WT_CURSOR *cursor)
         WT_TRET((*cp)->reset(*cp));
     }
 
+    /*
+     * The bounded cursor API clears bounds on external calls to cursor->reset. We determine this by
+     * guarding the call to cursor bound reset with the API_USER_ENTRY macro. Doing so prevents
+     * internal API calls from resetting cursor bounds unintentionally, e.g. cursor->remove.
+     */
+    if (API_USER_ENTRY(session))
+        __wt_cursor_bound_reset(cindex->child);
```

Comment:
Does the index cursor have colgroups like the table cursor?

Reply:
The table cursor resets bounds on all child cursors:
```
    if (API_USER_ENTRY(session))
        for (i = 0, cp = ctable->cg_cursors; i < WT_COLGROUPS(ctable->table); i++, cp++)
            __wt_cursor_bound_reset(*cp);
```

Reply:
Umm yes the index cursors does open cursors for each colgroup. However index cursors does not apply bounds to colgroups because the index child cursor does the required bounded operation first always. Therefore index cursors is only applied for the child cursor.